rotkehlchen/tests/exchanges/test_bitfinex.py

```diff
@@ -0,0 +1,1376 @@
+import warnings as test_warnings
+from contextlib import ExitStack
+from datetime import datetime
+from http import HTTPStatus
+from unittest.mock import call, patch
+
+import pytest
+from requests import Response
+
+from rotkehlchen.accounting.structures import Balance
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.assets.converters import (
+    BITFINEX_EXCHANGE_TEST_ASSETS,
+    UNSUPPORTED_BITFINEX_ASSETS,
+    asset_from_bitfinex,
+)
+from rotkehlchen.errors import (
+    RemoteError,
+    SystemClockNotSyncedError,
+    UnknownAsset,
+    UnsupportedAsset,
+)
+from rotkehlchen.exchanges.bitfinex import (
+    API_KEY_ERROR_CODE,
+    API_KEY_ERROR_MESSAGE,
+    API_RATE_LIMITS_ERROR_MESSAGE,
+    API_SYSTEM_CLOCK_NOT_SYNCED_ERROR_CODE,
+    Bitfinex,
+    CurrencyMapResponse,
+)
+from rotkehlchen.exchanges.data_structures import AssetMovement, Trade, TradeType
+from rotkehlchen.fval import FVal
+from rotkehlchen.tests.utils.mock import MockResponse
+from rotkehlchen.typing import (
+    AssetAmount,
+    AssetMovementCategory,
+    Fee,
+    Location,
+    Price,
+    Timestamp,
+    TradePair,
+)
+
+
+def test_name():
+    exchange = Bitfinex('a', b'a', object(), object())
+    assert exchange.name == str(Location.BITFINEX)
+
+
+def test_bitfinex_exchange_assets_are_known(mock_bitfinex):
+    """This tests only exchange (trades) assets (not margin, nor futures ones).
+    """
+    currencies_response = mock_bitfinex._query_currencies()
+    if currencies_response.success is False:
+        response = currencies_response.response
+        test_warnings.warn(UserWarning(
+            f'Failed to request {mock_bitfinex.name} currencies list. '
+            f'Response status code: {response.status_code}. '
+            f'Response text: {response.text}. Xfailing this test',
+        ))
+        pytest.xfail('xfailing this test')
```

Comment:
Add the reason for the xfail too. A small sentence would be enough. Same below. Not sure if warnings are shown next to the test failure.

Reply:
I have not added the sentence as the reason `xfail` is not shown unless we run pytest in "report-on-xfail" mode (`-rx` option). I will add it but the warning log should is very descriptive (reason + "Xfailing this test").